### PR TITLE
fix(TDI-42608): Salesforce bulk query batch job can't be closed when …

### PR DESCRIPTION
**Pull request by vizotenko-talend**
…error throw.

**What is the current behavior?** (You can also link to an open issue here)

**What is the new behavior?**

**Please check if the PR fulfills these requirements**

- [ ] The commit message follows Talend standard
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features) ?
- [ ] The code coverage on new code >75%
- [ ] The new code does not introduce new technical issues (sonar / eslint)

**What kind of change does this PR introduce?**

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build / CI related changes
- [ ] Other... Please describe:

**Does this PR introduce a breaking change?**

- [ ] Yes
- [ ] No

If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...

**Other information**:

### CODE REVIEW

**Comment by build-talend-common** [comment]
Pull request test result: [failure](https://ci-common.datapwn.com/job/components_pull-request_master/31/) [:x:](https://ci-common.datapwn.com/job/components_pull-request_master/31/)

File: components/components-salesforce/components-salesforce-runtime/src/main/java/org/talend/components/salesforce/runtime/SalesforceBulkQueryInputReader.java [hunk 1]
```diff
@@ -116,6 +117,9 @@ protected void executeSalesforceBulkQuery() throws IOException, ConnectionExcept
             bulkRuntime.doBulkQuery(getModuleName(), queryText);
         } catch (AsyncApiException | InterruptedException | ConnectionException e) {
             throw new IOException(e);
+        } catch (TalendRuntimeException e){
```
Comment (jzhao-talend) [inline components/components-salesforce/components-salesforce-runtime/src/main/java/org/talend/components/salesforce/runtime/SalesforceBulkQueryInputReader.java]: i think this only a fix part of the issue. only resolve when got exception during execute bulk query.
also need consider when exuecute the query successful, exception throw during extract query result:
eg. https://github.com/Talend/components/blob/c849daeb444dd7fec22b137eb0b3494bca1d2531/components/components-salesforce/components-salesforce-runtime/src/main/java/org/talend/components/salesforce/runtime/SalesforceBulkQueryInputReader.java#L69

Comment (vizotenko-talend) [reply components/components-salesforce/components-salesforce-runtime/src/main/java/org/talend/components/salesforce/runtime/SalesforceBulkQueryInputReader.java]: done

**Comment by build-talend-common** [comment]
Pull request test result: [failure](https://ci-common.datapwn.com/job/components_pull-request_master/32/) [:x:](https://ci-common.datapwn.com/job/components_pull-request_master/32/)

File: components/components-salesforce/components-salesforce-runtime/src/main/java/org/talend/components/salesforce/runtime/SalesforceBulkQueryInputReader.java [hunk 2]
```diff
@@ -66,7 +67,12 @@ public boolean start() throws IOException {
...
     public boolean advance() throws IOException {
-        currentRecord = bulkResultSet.next();
+        try {
+            currentRecord = bulkResultSet.next();
+        } catch (IOException e){
+            close();
+            throw e;
+        }
         if (currentRecord == null) {
             return retrieveNextResultSet();
```
Comment (jzhao-talend) [inline components/components-salesforce/components-salesforce-runtime/src/main/java/org/talend/components/salesforce/runtime/SalesforceBulkQueryInputReader.java]: this seems also need to include

Comment (vizotenko-talend) [reply components/components-salesforce/components-salesforce-runtime/src/main/java/org/talend/components/salesforce/runtime/SalesforceBulkQueryInputReader.java]: done

**Comment by build-talend-common** [comment]
Pull request test result: [failure](https://ci-common.datapwn.com/job/components_pull-request_master/33/) [:x:](https://ci-common.datapwn.com/job/components_pull-request_master/33/)

**Comment by build-talend-common** [comment]
Pull request test result: [success](https://ci-common.datapwn.com/job/components_pull-request_master/34/) [:star:](https://ci-common.datapwn.com/job/components_pull-request_master/34/) 
[Sonar results](https://sonar.datapwn.com/overview?id=org.talend.components%3Acomponents%3Avizotenko%2FTDI-42608)

**Pull request merged**